fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/api/event/registry/FabricRegistryBuilder.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.registry;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import net.minecraft.util.registry.Registry;
+
+import net.fabricmc.fabric.impl.registry.sync.FabricRegistry;
+
+/**
+ * Used to create custom registry's, with specified registry attributes.
+ *
+ * <pre>
+ *     MutableRegistry<String> exampleRegistry = FabricRegistryBuilder.create(new SimpleRegistry<String>())
```

Comment:
I personally find it a bit odd to pass registry into a builder.

Reply:
Yeah, it is slightly odd. I could keep this but add createSimple and createDefault?

Reply:
Yeah that seems like a reasonable approach. Maybe make the method where you can pass a custom registry through something like `from`.

I'd `createDefaulted` is a bit more descriptive for the second name

Reply:
`{@literal}` please for `<` and `>`, or use `&lt;` for `<` and `&gt;` for `>`.

Reply:
Or you can use a `{@code}` tag as well, see [this](https://github.com/FabLabsMC/ScreenHandlers/blob/42a5360fb600625ddb161f3abb1cdcc2b85dc824/src/main/java/io/github/fablabsmc/fablabs/api/screenhandler/v1/FabricScreenHandlerTypes.java#L35).